packages/@glimmer/runtime/lib/references/curry-value.ts

```diff
@@ -0,0 +1,68 @@
+import { DEBUG } from '@glimmer/env';
+import {
+  CapturedArguments,
+  CurriedType,
+  Dict,
+  Maybe,
+  Option,
+  Owner,
+  RuntimeResolver,
+} from '@glimmer/interfaces';
+import { createComputeRef, Reference, valueForRef } from '@glimmer/reference';
+import { expect } from '@glimmer/util';
+import { curry, isCurriedType } from '../curried-value';
+
+export default function createCurryRef(
+  type: CurriedType,
+  inner: Reference,
+  owner: Owner,
+  args: Option<CapturedArguments>,
+  resolver: RuntimeResolver,
+  isStrict: boolean
+) {
+  let lastValue: Maybe<Dict> | string, curriedDefinition: object | string | null;
+
+  return createComputeRef(() => {
+    let value = valueForRef(inner) as Maybe<Dict> | string;
+
+    if (value === lastValue) {
+      return curriedDefinition;
+    }
+
+    if (isCurriedType(value, type)) {
+      curriedDefinition = args ? curry(type, value, owner, args) : args;
+    } else if (type === CurriedType.Component && typeof value === 'string' && value) {
+      // Only components should enter this path, as helpers and modifiers do not
+      // support string based resolution
+
+      if (DEBUG && isStrict) {
+        if (isStrict) {
```

Comment:
This double `isStrict` check seems odd. Isn't the outer `if (DEBUG && isStrict) {` enough?

Reply:
yeah, that was a mistake, the `isStrict` should only be on the inside, and the whole block should execute regardless of strictness (only one assertion is related to strict)